src/contexts/PageContext.js

```diff
@@ -0,0 +1,62 @@
+import { createContext, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+export const PageContext = createContext();
+
+export const PageProvider = ({ children }) => {
+  const navigate = useNavigate();
+
+  const [currentPage, setCurrentPage] = useState("prompt");
+  const [progressBarIndex, setProgressBarIndex] = useState(0);
+
+  const handleNext = () => {
+    if (progressBarIndex < 3) {
+      setProgressBarIndex((prevIndex) => prevIndex + 1);
+    }
+
+    if (currentPage === "prompt") {
```

Comment:
Naming pages with string is not wrong. But you an also set the page to some array and then use the index to manage the progress of that array. 

like const progressPageArray = [ , "prompt", "onboarding1", ... ]

Reply:
This approach seems cleaner and with less if/else blocks in my handleNext and handleBack functions.